Cookbook Guide for Mirabella Genio Smart Bulbs

cryptelli opened the pull request:
## Description:
Cookbook guide for Mirabella Genio Smart Bulbs with configuration examples.

Co-Authored-By: MartinP <mplinuxgeek@users.noreply.github.com>

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#<esphome PR number goes here>
**Pull request in [esphome-core](https://github.com/esphome/esphome-core) with C++ framework changes (if applicable):** esphome/esphome-core#<esphome-core PR number goes here>

## Checklist:

  - [x] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.

cryptelli committed:
Cookbook Guide for Mirabella Genio Smart Bulbs

Cookbook guide for Mirabella Genio Smart Bulbs with configuration examples.

Co-Authored-By: MartinP <mplinuxgeek@users.noreply.github.com>

OttoWinter committed:
✏Some changes

OttoWinter committed:
Update mirabella-genio-bulb.rst

OttoWinter commented on cookbook/mirabella-genio-bulb.rst:
```diff
@@ -0,0 +1,175 @@
+    :align: center
+    :width: 50.0%
+
+Originally intended to be used with their companion app once flashed using `tuya-convert <https://github.com/ct-Open-Source/tuya-convert>`__ ESPHome generated firmware can be uploaded allowing you to control the bulbs via Home Assistant.
```
We should have a quick note here about how to use tuya-convert with ESPHome. AFAIK the setup is a bit more difficult, as you need to first generate the bin and the flash it from tuya-convert. Doesn't have to be long and we can link to other tutorials.

OttoWinter commented on cookbook/mirabella-genio-bulb.rst:
```diff
@@ -0,0 +1,175 @@
+      on_boot:
+        priority: 100 # Highest priority, ensures light turns on without delay.
+        then:
+          - light.turn_on: light
```
So the physical switch on the device turns off the whole ESP?

cryptelli replied:
Sorry, not sure what you mean here 😕.

There is no physical switch apart from the one that controls the light, without the `on_boot` the light doesn't automatically turn on if the switch is toggled.

OttoWinter commented on cookbook/mirabella-genio-bulb.rst:
```diff
@@ -0,0 +1,175 @@
+    logger:
+
+    web_server:
+      port: 80
```
Let's not enable the web server by default unless there's a specific reason to do so.

OttoWinter commented on cookbook/mirabella-genio-bulb.rst:
```diff
@@ -0,0 +1,175 @@
+
+Dimmable Warm/Cool White Bulbs
+------------------------------
+
```
Please add some comments here as to what this config is doing.

Can be quite short too, just mention that `esp8266_pwm` is used here to control the light, and how they're connected with IDs.

OttoWinter commented on cookbook/mirabella-genio-bulb.rst:
```diff
@@ -0,0 +1,175 @@
+
+Thanks to the `existing work <https://github.com/arendst/Sonoff-tasmota/wiki/Mirabella-Genio-Bulb>`__ by `@so3n <https://github.com/arendst/Sonoff-tasmota/wiki/Mirabella-Genio-Bulb>`__ which these adaptions created by `@cryptelli <https://community.home-assistant.io/u/cryptelli>`__ and `@bircoe <https://community.home-assistant.io/u/bircoe>`__ are based on.
+
+Dimmable Warm/Cool White Bulbs
```
This section shows a monochromatic light? Let's call this section brightness-only or even monochromatic bulbs

OttoWinter commented on cookbook/mirabella-genio-bulb.rst:
```diff
@@ -0,0 +1,175 @@
+        id: output_component1
+        pin: GPIO13
+
+Tunable Warm/Cool White Bulbs
```
Let's not use `tunable` here, the word could mean anything here. Maybe something more like cold white + warm white bulbs

OttoWinter commented on cookbook/mirabella-genio-bulb.rst:
```diff
@@ -0,0 +1,175 @@
+        cold_white_color_temperature: 6500 K
+        warm_white_color_temperature: 2700 K
+
+RGBW Colour Bulbs
```
```suggestion
RGBW Color Bulbs
```

American english :)

cryptelli replied:
:smile:

OttoWinter commented on cookbook/mirabella-genio-bulb.rst:
```diff
@@ -0,0 +1,175 @@
+-------------------------
+
+You can now add your bulb to home assistant via the configurations page, look for 'ESPHome' under the Integrations option and click 'Configure'.
+
```
If possible, add a screenshot here of what it looks like in HA - best with the RGB version and with the color picker open.

cryptelli commented:
Boom 💥. Didn't intend on making more work for you! You've made very valid points which I'll address 👍.

cryptelli committed:
Major rewrite of the documentation

Addresses points raised by @OttoWinter from initial pull request and adds detailed documentation surrounding the flashing procedure for tuya-convert :thumbsup:.

OttoWinter committed:
Update Makefile

OttoWinter committed:
Lint

OttoWinter commented:
Sorry for the long delay, I was quite busy the last few days.

Thanks for the detailed guide. The text looks good to me.

However, this guide should be created against the `current` branch (not next) as it's not a new esphome feature being documented, just a documentation update.

So to fix this, you need to go to the git repo

```bash
git checkout dev
git pull
git checkout -b mirabella-genio
# Copy changes here
git add .
git commit -m "Commit Message"
git push
```

cryptelli commented:
All good, likewise. I misunderstood the meaning of the next branch. I just changed the branch the pull request is based on above, hopefully that's suitable?

Regarding the travis log, as I'm fairly new to reStructuredText so not sure how to address the line length issues in my lists? Also the issue it has with links not being relative are in my code comments which won't be helpful to the end user if changed.

cryptelli committed:
Address linting errors

cryptelli committed:
Address additional linting errors

cryptelli commented:
I've addressed most style errors, apart from the comments in my code blocks.

OttoWinter committed:
Lint esphome.io links

I just removed them, a user that wants to look it up can go manually to the link. It's not too hard to find.

OttoWinter commented:
Ok, thanks @cryptelli!

The branch you created the PR from was still `next` so commits that were not part of this PR were showing up (and all recent commits from next would go into current if I merged, which would break a bunch of things). So I reset the branch to current and cherry picked all commits that were part of this PR.

I also updated the contributing section to give some more info how to use branches in git (ideally you wouldn't create PRs from your current/next branch but from a new feature branch).

I will wait for the PR checks to pass and then merge it.

cryptelli commented:
Sorry @OttoWinter... appreciate you merging this PR after all that!

The expanded documentation is much clearer. I'll be sure to follow the correct method on my next PR.
